primitives/application-crypto/src/lib.rs

```diff
@@ -373,11 +374,12 @@ macro_rules! app_crypto_public_common {
 	};
 }
 
-/// Implements traits for the public key type if `feature = "std"` is enabled.
-#[cfg(feature = "std")]
+/// Implements traits for the public key type if `feature = "serde"` and `feature = "full_crypto"`
+/// are enabled.
+#[cfg(all(feature = "serde", feature = "full_crypto"))]
```

Comment:
Why again serde and full crypto? Especially as the name does not hints this. I think we should only have `feature = "serde"` here